uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/service/UniProtEntryService.java

```diff
@@ -102,7 +105,9 @@ public QueryResult<UniProtKBEntry> search(SearchRequest request) {
         QueryResult<UniProtDocument> results =
                 repository.searchPage(solrRequest, request.getCursor());
         List<ReturnField> fields = OutputFieldsParser.parse(request.getFields(), returnFieldConfig);
-        return resultsConverter.convertQueryResult(results, fields);
+        ProblemPair warning = getLeadingWildcardIgnoredWarning(request.getQuery());
```

Comment:
I think this logic needs to be in the parent/generic class, because the leading wildcard logic is used for everything, not only UniProt, we need to make sure we add the warning for everything, Does it make sense to you?